// vue-joystick-component/dist/types.mjs
export var JoystickComponent;
((JoystickComponent2) => {
  let Shape;
  ((Shape2) => {
    Shape2["Circle"] = "circle";
    Shape2["Square"] = "square";
  })(Shape = JoystickComponent2.Shape || (JoystickComponent2.Shape = {}));
  let InteractionEvents;
  ((InteractionEvents2) => {
    InteractionEvents2["PointerDown"] = "pointerdown";
    InteractionEvents2["PointerMove"] = "pointermove";
    InteractionEvents2["PointerUp"] = "pointerup";
    InteractionEvents2["FollowStop"] = "FOLLOW_STOP";
  })(InteractionEvents = JoystickComponent2.InteractionEvents || (JoystickComponent2.InteractionEvents = {}));
  let RadianQuadrantBinding;
  ((RadianQuadrantBinding2) => {
    RadianQuadrantBinding2[RadianQuadrantBinding2["TopRight"] = 2.35619449] = "TopRight";
    RadianQuadrantBinding2[RadianQuadrantBinding2["TopLeft"] = -2.35619449] = "TopLeft";
    RadianQuadrantBinding2[RadianQuadrantBinding2["BottomRight"] = 0.785398163] = "BottomRight";
    RadianQuadrantBinding2[RadianQuadrantBinding2["BottomLeft"] = -0.785398163] = "BottomLeft";
  })(RadianQuadrantBinding = JoystickComponent2.RadianQuadrantBinding || (JoystickComponent2.RadianQuadrantBinding = {}));
})(JoystickComponent || (JoystickComponent = {}));
